### PR TITLE
Added keccak256/ripemd160/sha2-256 system contract for ewasm

### PR DESCRIPTION
**Pull request by sxysun-zz**
This resolves the issue in the solang repo: https://github.com/hyperledger-labs/solang/issues/268

In ewasm design doc, it writes that the calls to hash functions should be implemented by calls to external contracts (system contracts). But currently, solang uses its own hash function implementation, so we want to add some resolvers to the wasm VM in burrow to support that.

### CODE REVIEW

**Review by gregdhill — APPROVED** [review]

**Comment by gregdhill** [comment]
Please sign the DCO, other test failures are unrelated to this PR. Otherwise, LGTM!

**Comment by seanyoung** [comment]
@sxysun looks great, thank you.

Please can you remove the .DS_Store and the execution/.DS_Store files; they should not be checked in.

**Review by seanyoung — CHANGES_REQUESTED** [review]
Please can you remove the .DS_Store and the execution/.DS_Store files; they should not be checked in.

File: execution/wasm/wasm.go [hunk 1]
```diff
@@ -73,6 +74,35 @@ func (e *execContext) ResolveFunc(module, field string) exec.FunctionImport {
 	}
 
 	switch field {
+	case "call":
+		return func(vm *exec.VirtualMachine) int64 {
+			// gas := int(uint64(vm.GetCurrentFrame().Locals[0]))
```
Comment (seanyoung) [inline execution/wasm/wasm.go:79]: Ideally the call function should be implemented properly and then the precompiles in https://github.com/hyperledger/burrow/blob/master/execution/native/precompiles.go can be called.

**Comment by seanyoung** [comment]
@silasdavis to implement proper calls:

Add the lines: https://github.com/hyperledger/burrow/blob/6eaa052f6962f441bf8a469bd1b83026e7708d47/execution/wasm/wasm.go#L77-L88 (you may want to give `address` the type `crypto.Address`).

The call data is in vm.Memory[dataPtr:dataPtr+dataLen]
The call must then be dispatched and the returndata stored in `e.returnData` and the appropriate return code passed on.

That should be all to make external calls work.

Once that is in place, the precompiles can be used for hashing and the code here is no longer needed.

**Comment by silasdavis** [comment]
This did take a little while for me to come around to, but I have began to implement (really to finish off) cross-engine calls in this PR: https://github.com/hyperledger/burrow/pull/1421

As Sean says once of the things this will allow is for WASM code to call the precompiles (or any of the Go native contracts). Once we have that up and running we can close this in favour of that. Thanks for pointing this out @sxysun

**Pull request closed without merging**